### PR TITLE
WIP: GMime

### PR DESCRIPTION
**Pull request by vhdirk**
I've been working on bindings for GMime.

Currently, the gmime-sys crate builds, and allows the top-layer crate (https://github.com/vhdirk/gmime-rs) to build against it.

Even though this PR is currently not suitable for merging. What needs to be done in order for it to get merged?

### CODE REVIEW

File: gmime-sys/Cargo.lock [hunk 1]
```diff
@@ -0,0 +1,310 @@
+[[package]]
```
Comment (EPashkin) [inline gmime-sys/Cargo.lock]: This file better be removed from PR

File: conf/gir-gmime.toml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+external_libraries = [
+   "GLib",
+   "GObject",
+   "Gio",
+]
+
+
+[[object]]
+name = "GMime._GMimeFilterGZipPrivate"
+status = "manual"
```
Comment (EPashkin) [inline conf/gir-gmime.toml]: This better done with
```
manual = [
    "GMime._GMimeFilterGZipPrivate",
    "GMime._GMimeParserPrivate",
]
```

File: gmime-sys/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+[build-dependencies]
+pkg-config = "0.3.7"
+
+[dependencies]
+bitflags = "1.0"
+libc = "0.2"
+iconv = { git = 'https://github.com/y-stm/rust-iconv' }
```
Comment (EPashkin) [inline gmime-sys/Cargo.toml]: I don't see use any types from iconv, so this IMHO unneeded.

**Comment by EPashkin** [comment]
I not against adding `GMime-3.0.gir` to gir-files as long as it can normally regenerated,
but IMHO this gmime-sys need separate repo or included in https://github.com/vhdirk/gmime-rs as we do for cairo

**Comment by vhdirk** [comment]
I can include this in gmime-rs (I'll have a look at cairo). That may also make it easier for me to develop is, since maintaining 3 repos is a bit of a hassle.

**Comment by vhdirk** [comment]
I'm moving this to the main gmime-rs repo: https://github.com/vhdirk/gmime-rs.
That's actually far easier to maintain. Any conrtibutions would be more than welcome! (How do you do a pull request for an entire repo?)

**Comment by EPashkin** [comment]
> How do you do a pull request for an entire repo

Same as this PR ;)
Or you meant something different?

**Comment by vhdirk** [comment]
I meant that at some point the mime-rs repo could be eligible for moving into the gtk-rs namespace?

**Comment by EPashkin** [comment]
Well `gmime` seems works in Windows too.
So I don't see big problem with transferring when it ready.
@GuillaumeGomez, @sdroege what you think about this?

**Comment by GuillaumeGomez** [comment]
Are we talking about a repository or a type?

**Comment by EPashkin** [comment]
About repo

**Comment by sdroege** [comment]
I don't really have an opinion. Go for it if someone wants to maintain it :)

Is anybody actually still using gmime, is it still maintained?

**Comment by vhdirk** [comment]
Not sure if it's still actively maintained. I think Geary uses it?
I was considering a rust-only solution for a mail client I'm building, but the eco system in that aspect is still rather lacking. GMime seems far more capable. I'm not sure if I'll be able to maintain it in the long run, but gmime seems to evolve _really_ slowly, if at all.

**Pull request closed without merging**